### PR TITLE
Drop unused generated clients

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area scalability dev-productivity
/kind cleanup

**What this PR does / why we need it**:

This PR drops some unused generated client code (extensions and operations group).
It also drops some unused clients (`Garden{Core,SeedManagement,Operations}` and `API{Registration,Extension}`) from `kubernetes.Interface`.

All of this is dropped in favor of (cached) controller-runtime clients to simplify/unify our code and leverage the clients' caches wherever possible.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/2414

**Special notes for your reviewer**:

`kubernetes.Interface.Kubernetes()` is not dropped in this PR, as there are still a lot of usages.
This can be tackled in a future PR.

The generated operations clients might be needed in the future, when we add some checks in admission plugins (e.g. for `Bastion`s). I decided to drop them nevertheless for the sake of simplification. We can still add them back with a few lines later on if needed.

**Release note**:

```other operator

```

### CODE REVIEW

File: pkg/utils/kubernetes/managedseed.go [hunk 1]
```diff
@@ -44,3 +45,20 @@ func GetManagedSeed(ctx context.Context, seedManagementClient gardenseedmanageme
 	}
 	return &managedSeedList.Items[0], nil
 }
+
+// GetManagedSeedWithReader gets the ManagedSeed resource for the given shoot namespace and name,
+// by searching for all ManagedSeeds in the shoot namespace that have spec.shoot.name set to the shoot name.
+// If no such ManagedSeeds are found, nil is returned.
+func GetManagedSeedWithReader(ctx context.Context, r client.Reader, shootNamespace, shootName string) (*seedmanagementv1alpha1.ManagedSeed, error) {
```
Comment (rfranzke) [inline pkg/utils/kubernetes/managedseed.go:52]: Can the old function (`GetManagedSeed`) be deleted?

**Review by rfranzke — COMMENTED** [review]
Love it! I assume you plan to remove the other clients with subsequent PRs?

File: pkg/utils/kubernetes/managedseed.go (hunk 1 above)
Comment (timebertt) [reply pkg/utils/kubernetes/managedseed.go:52]: Nope, that one is still used by admission plugins.

**Comment by timebertt** [comment]
> I assume you plan to remove the other clients with subsequent PRs?

That won't be possible so easily, as they are used in admission plugins.
Currently, I'm not sure, if or how we can switch to controller-runtime clients there (tracked in https://github.com/gardener/gardener/issues/2414).

However, what is possible and planned is to remove usages of `kubernetes.Interface.Kubernetes()` (in all gcm/gardenlet code).

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**